luabind/detail/object_rep.hpp

```diff
@@ -97,8 +97,14 @@ namespace luabind { namespace detail
 
         void operator=(object_rep const&);
 
+#ifdef LUAI_MAXALIGN
+        struct cD { char c; union { LUAI_MAXALIGN; } u; };
+#else
+        struct cD { char c; union { lua_Number n; double u; void *s; lua_Integer i; long l; } u; };
```

Comment:
I believe `long` is not quite right here, as it will only be 32 bit on Windows. But lua_Integer should cover that though.

Reply:
Yeah, that's just matching the LUAI_MAXALIGN macro, since it was only added in 5.4.